addons/mail/models/mail_channel.py

```diff
@@ -88,21 +88,26 @@ def _generate_random_token(self):
         ('groups', 'Selected group of users')], string='Privacy',
         required=True, default='groups',
         help='This group is visible by non members. Invisible groups can add members through the invite button.')
-    group_public_id = fields.Many2one('res.groups', string='Authorized Group',
-                                      default=lambda self: self.env.ref('base.group_user'))
+    group_public_id = fields.Many2one('res.groups', string='Authorized Group', compute='_compute_group_public_id', readonly=False, store=True)
```

Comment:
Ideally this should actually be readonly=True when channel type is invalid to prevent the user from inputting invalid values that will be refused by the constraint. Could possibly add the domain here in python, or at least in the form view.